packages/mdc-segmented-button/segment/foundation.ts

```diff
@@ -42,22 +42,37 @@ export class MDCSegmentedButtonSegmentFoundation extends MDCFoundation<MDCSegmen
   }
 
   isSelected(): boolean {
-    return false;
+    return this.adapter.hasClass('mdc-segmented-button__segment--selected');
   }
 
   setSelected() {
-    return;
+    this.adapter.addClass('mdc-segmented-button__segment--selected');
+    this.setAriaAttr('true');
   }
 
   setUnselected() {
-    return;
+    this.adapter.removeClass('mdc-segmented-button__segment--selected');
+    this.setAriaAttr('false');
   }
 
   getSegmentId(): string {
-    return '';
+    return this.adapter.getAttr('id');
   }
 
-  handleClick() {
-    return;
+  handleClick(): void {
+    if (this.isSelected()) {
```

Comment:
Are we adding logic only for multi-select in this PR?

Reply:
Yes. The child segment changes status automatically when it's clicked and notifies the parent segment-button. Then the parent-segment button handles the single-/multi-select logic and modifies the child segments as needed.

Reply:
Single select segmented button can't be toggled. It should behave similar to radio group. For example, click on segment button that is already selected should not unselect it (if it is single select).

I think segment foundation can be of two types (one that toggles & another that does not toggle)

Reply:
I had it to unselecting a segment in a single select segmented button would unselect it, then reselect it to be functionally the same as not being unselected, but I can see why this could have drawbacks. Since the Adapter already has a isSingleSelect method I can use that to keep the segment from unselecting.

Reply:
This can be handled by the parent; it fires an event upward and the parent can do the logic of whether it should unselect or not by calling the adapter to select this one. That would obviate the need for `isSingleSelect`

Reply:
Selecting a segment which is already selected in Single select mode does should not change color and also should not unselect it. But it can trigger an event.

I would keep the save behaviour as [radio group](https://www.w3.org/TR/2016/WD-wai-aria-practices-1.1-20160317/examples/radio/radio.html).

A segment can be toggled only for multi-select. I think it makes sense to set this boolean from parent depending on single or multi-select.

Reply:
`isSingleSelect` is also used to determine which aria-pattern to use, which seemed for appropriate for the child to control but could be handled by the parent, thus removing the need for `isSingleSelect`.
If we keep `isSingleSelect` and the child doesn't allow itself to be unselected when the parent is single-select then the parent doesn't have to check whether to re-select the child. The parent only has to unselect the previously selected child.